controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -633,20 +633,25 @@ private void closeConnection(Reply reply, RawClient client) {
     @VisibleForTesting
     <T extends Request & WireCommand> void processAndRethrowException(long requestId, T request, Throwable e) {
         Throwable unwrap = Exceptions.unwrap(e);
+        WireCommandFailedException ex = null;
         if (unwrap instanceof ConnectionFailedException || unwrap instanceof ConnectionClosedException) {
-            log.warn(requestId, "Connection dropped");
-            throw new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.ConnectionFailed);
+            ex = new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.ConnectionFailed);
+            log.warn(requestId, "Connection dropped: {}", ex.getMessage());
+            throw ex;
         } else if (unwrap instanceof AuthenticationException) {
-            log.warn(requestId, "Authentication Exception");
-            throw new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.AuthFailed);
+            ex = new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.AuthFailed);
+            log.warn(requestId, "Authentication Exception: {}", ex.getMessage());
+            throw ex;
         } else if (unwrap instanceof TokenExpiredException) {
-            log.warn(requestId, "Token expired");
-            throw new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.AuthFailed);
+            ex = new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.AuthFailed);
+            log.warn(requestId, "Token expired: {}", ex.getMessage());
+            throw ex;
         } else if (unwrap instanceof TimeoutException) {
-            log.warn(requestId, "Request timed out.");
-            throw new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.ConnectionFailed);
+            ex = new WireCommandFailedException(request.getType(), WireCommandFailedException.Reason.ConnectionFailed);
+            log.warn(requestId, "Request timed out: {}", ex.getMessage());
```

Comment:
Is it possible to give more details of the request for which no response (in 120 seconds) from the segment store was observed ? I think we can log go ahead and log the complete request here.

Reply:
Ok done